### ISSUE TITLE
esbuild: Failed to install correctly

### ISSUE THREAD

**Issue by mvysny** [vaadin/skeleton-starter-flow#519]
Trying to run this project master (which at the moment uses Vaadin 22.0.0) in my Tomcat fails with

```
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - [webpack-dev-server] On Your Network (IPv4): http://192.168.0.17:36629/
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - [webpack-dev-server] Content not from webpack is served from '/home/mavi/work/vaadin/starters/14/skeleton-starter-flow/target/classes/META-INF/VAADIN/webapp, /home/mavi/work/vaadin/starters/14/skeleton-starter-flow/src/main/webapp' directory
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - /home/mavi/work/vaadin/starters/14/skeleton-starter-flow/node_modules/.pnpm/esbuild@0.12.29/node_modules/esbuild/bin/esbuild:2
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - throw new Error(`esbuild: Failed to install correctly
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - ^
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - 
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - Error: esbuild: Failed to install correctly
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - 
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - Make sure you don't have "ignore-scripts" set to true. You can check this with
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - "npm config get ignore-scripts". If that returns true you can reset it back to
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - false using "npm config set ignore-scripts false" and then reinstall esbuild.
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - 
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - If you're using npm v7, make sure your package-lock.json file contains either
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - "lockfileVersion": 1 or the code "hasInstallScript": true. If it doesn't have
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - either of those, then it is likely the case that a known bug in npm v7 has
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - corrupted your package-lock.json file. Regenerating your package-lock.json file
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - should fix this issue.
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - 
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker -     at Object.<anonymous> (/home/mavi/work/vaadin/starters/14/skeleton-starter-flow/node_modules/.pnpm/esbuild@0.12.29/node_modules/esbuild/bin/esbuild:2:7)
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker -     at Module._compile (internal/modules/cjs/loader.js:999:30)
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker -     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker -     at Module.load (internal/modules/cjs/loader.js:863:32)
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker -     at Function.Module._load (internal/modules/cjs/loader.js:708:14)
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker -     at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:60:12)
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker -     at internal/main/run_main_module.js:17:47
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - 
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - ERROR in ./generated/vaadin.ts
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - Module build failed (from ../node_modules/.pnpm/esbuild-loader@2.15.1_webpack@4.46.0/node_modules/esbuild-loader/dist/index.js):
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - Error: The service was stopped
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker -     at /home/mavi/work/vaadin/starters/14/skeleton-starter-flow/node_modules/.pnpm/esbuild@0.12.29/node_modules/esbuild/lib/main.js:1258:29
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker -     at /home/mavi/work/vaadin/starters/14/skeleton-starter-flow/node_modules/.pnpm/esbuild@0.12.29/node_modules/esbuild/lib/main.js:611:9
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker -     at Socket.afterClose (/home/mavi/work/vaadin/starters/14/skeleton-starter-flow/node_modules/.pnpm/esbuild@0.12.29/node_modules/esbuild/lib/main.js:589:7)
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker -     at Socket.emit (events.js:326:22)
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker -     at endReadableNT (_stream_readable.js:1241:12)
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker -     at processTicksAndRejections (internal/process/task_queues.js:84:21)
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - 
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - ERROR in ../target/sw.ts
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - Module build failed (from ../node_modules/.pnpm/esbuild-loader@2.15.1_webpack@4.46.0/node_modules/esbuild-loader/dist/index.js):
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - Error: The service was stopped
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker -     at /home/mavi/work/vaadin/starters/14/skeleton-starter-flow/node_modules/.pnpm/esbuild@0.12.29/node_modules/esbuild/lib/main.js:1258:29
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker -     at /home/mavi/work/vaadin/starters/14/skeleton-starter-flow/node_modules/.pnpm/esbuild@0.12.29/node_modules/esbuild/lib/main.js:611:9
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker -     at Socket.afterClose (/home/mavi/work/vaadin/starters/14/skeleton-starter-flow/node_modules/.pnpm/esbuild@0.12.29/node_modules/esbuild/lib/main.js:589:7)
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker -     at Socket.emit (events.js:326:22)
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker -     at endReadableNT (_stream_readable.js:1241:12)
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker -     at processTicksAndRejections (internal/process/task_queues.js:84:21)
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - 
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - ERROR in Can't find self.__WB_MANIFEST in your SW source.
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - [build-status] 3 errors and 0 warnings were reported.
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - [build-status] : Failed to compile.
[ForkJoinPool.commonPool-worker-19] INFO com.vaadin.base.devserver.AbstractDevServerRunner - Started Webpack. Time: 2695ms

------------------ Frontend compilation failed. ------------------

[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - Issues checking in progress...
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - No issues found.
/home/mavi/local/apache-tomcat-9.0.44/bin/catalina.sh stop
Disconnected from the target VM, address: '127.0.0.1:33705', transport: 'socket'
NOTE: Picked up JDK_JAVA_OPTIONS:  --add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.base/java.io=ALL-UNNAMED --add-opens=java.base/java.util=ALL-UNNAMED --add-opens=java.base/java.util.concurrent=ALL-UNNAMED --add-opens=java.rmi/sun.rmi.transport=ALL-UNNAMED
07-Dec-2021 12:59:41.898 INFO [main] org.apache.catalina.core.StandardServer.await A valid shutdown command was received via the shutdown port. Stopping the Server instance.
07-Dec-2021 12:59:41.898 INFO [main] org.apache.coyote.AbstractProtocol.pause Pausing ProtocolHandler ["http-nio-8080"]
07-Dec-2021 12:59:41.905 INFO [main] org.apache.catalina.core.StandardService.stopInternal Stopping service [Catalina]
07-Dec-2021 12:59:41.973 INFO [main] org.apache.coyote.AbstractProtocol.stop Stopping ProtocolHandler ["http-nio-8080"]
07-Dec-2021 12:59:42.003 INFO [main] org.apache.coyote.AbstractProtocol.destroy Destroying ProtocolHandler ["http-nio-8080"]
Disconnected from server
```

I've tried `mvn vaadin:dance` to no avail. I understand I could work around this issue by running `npm config set ignore-scripts false` but that's a security hole. Any tips on how to deal with this issue in a proper way please?

Environment: Ubuntu 21.10 x86-64, Vaadin 22.0.0, skeleton-starter-flow master, JDK 11

**Comment by mvysny** [comment]
Fix coming in https://github.com/vaadin/flow/pull/12512

**Comment by mvysny** [comment]
Not fixed in Vaadin 22.0.1

**Comment by mvysny** [comment]
The very same happen when I run the project via Maven+Jetty, via `mvn -C clean package jetty:run`:

```
...
[INFO] Scanning elapsed time=545ms
[INFO] Initializing AtmosphereFramework
[main] INFO com.vaadin.base.devserver.startup.DevModeStartupListener - Starting dev-mode updaters in /home/mavi/work/vaadin/starters/14/skeleton-starter-flow folder.
[main] INFO com.vaadin.flow.server.frontend.scanner.FullDependenciesScanner - Visited 118 classes. Took 57 ms.
[INFO] DefaultSessionIdManager workerName=node0
[INFO] No SessionScavenger set, using defaults
[INFO] node0 Scavenging every 600000ms
[main] INFO com.vaadin.flow.server.startup.ServletDeployer - Automatically deploying Vaadin servlet with name com.vaadin.flow.server.startup.ServletDeployer to /*
Vaadin application has been deployed and started to the context path "/".
[ForkJoinPool.commonPool-worker-3] INFO com.vaadin.flow.server.frontend.TaskUpdatePackages - Running `pnpm install` to resolve and optionally download frontend dependencies. This may take a moment, please stand by...
[INFO] Started o.e.j.m.p.JettyWebAppContext@159408a7{/,[file:///home/mavi/work/vaadin/starters/14/skeleton-starter-flow/src/main/webapp/, jar:file:///home/mavi/.m2/repository/org/webjars/npm/vaadin__vaadin-mobile-drag-drop/1.0.1/vaadin__vaadin-mobile-drag-drop-1.0.1.jar!/META-INF/resources, jar:file:///home/mavi/.m2/repository/com/vaadin/vaadin-select-flow/22.0.1/vaadin-select-flow-22.0.1.jar!/META-INF/resources, jar:file:///home/mavi/.m2/repository/com/vaadin/vaadin-menu-bar-flow/22.0.1/vaadin-menu-bar-flow-22.0.1.jar!/META-INF/resources, jar:file:///home/mavi/.m2/repository/com/vaadin/vaadin-grid-flow/22.0.1/vaadin-grid-flow-22.0.1.jar!/META-INF/resources, jar:file:///home/mavi/.m2/repository/com/vaadin/flow-push/9.0.1/flow-push-9.0.1.jar!/META-INF/resources, jar:file:///home/mavi/.m2/repository/com/vaadin/vaadin-renderer-flow/22.0.1/vaadin-renderer-flow-22.0.1.jar!/META-INF/resources, jar:file:///home/mavi/.m2/repository/com/vaadin/vaadin-lumo-theme/22.0.1/vaadin-lumo-theme-22.0.1.jar!/META-INF/resources, jar:file:///home/mavi/.m2/repository/com/vaadin/vaadin-messages-flow/22.0.1/vaadin-messages-flow-22.0.1.jar!/META-INF/resources, jar:file:///home/mavi/.m2/repository/com/vaadin/vaadin-text-field-flow/22.0.1/vaadin-text-field-flow-22.0.1.jar!/META-INF/resources, jar:file:///home/mavi/.m2/repository/com/vaadin/collaboration-engine/4.0.0/collaboration-engine-4.0.0.jar!/META-INF/resources, jar:file:///home/mavi/.m2/repository/com/vaadin/flow-dnd/9.0.1/flow-dnd-9.0.1.jar!/META-INF/resources, jar:file:///home/mavi/.m2/repository/com/vaadin/flow-client/9.0.1/flow-client-9.0.1.jar!/META-INF/resources, jar:file:///home/mavi/.m2/repository/com/vaadin/vaadin-combo-box-flow/22.0.1/vaadin-combo-box-flow-22.0.1.jar!/META-INF/resources, jar:file:///home/mavi/.m2/repository/com/vaadin/vaadin-date-picker-flow/22.0.1/vaadin-date-picker-flow-22.0.1.jar!/META-INF/resources, jar:file:///home/mavi/.m2/repository/com/vaadin/vaadin-time-picker-flow/22.0.1/vaadin-time-picker-flow-22.0.1.jar!/META-INF/resources, jar:file:///home/mavi/.m2/repository/com/vaadin/vaadin-grid-pro-flow/22.0.1/vaadin-grid-pro-flow-22.0.1.jar!/META-INF/resources, jar:file:///home/mavi/.m2/repository/com/vaadin/vaadin-material-theme/22.0.1/vaadin-material-theme-22.0.1.jar!/META-INF/resources, jar:file:///home/mavi/.m2/repository/org/webjars/npm/mobile-drag-drop/2.3.0-rc.2/mobile-drag-drop-2.3.0-rc.2.jar!/META-INF/resources, jar:file:///home/mavi/.m2/repository/com/vaadin/vaadin-virtual-list-flow/22.0.1/vaadin-virtual-list-flow-22.0.1.jar!/META-INF/resources, jar:file:///home/mavi/.m2/repository/com/vaadin/vaadin-iron-list-flow/22.0.1/vaadin-iron-list-flow-22.0.1.jar!/META-INF/resources, jar:file:///home/mavi/.m2/repository/com/vaadin/vaadin-context-menu-flow/22.0.1/vaadin-context-menu-flow-22.0.1.jar!/META-INF/resources],AVAILABLE}{file:///home/mavi/work/vaadin/starters/14/skeleton-starter-flow/src/main/webapp/}
[INFO] Started ServerConnector@670481b2{HTTP/1.1, (http/1.1)}{0.0.0.0:8080}
[INFO] Started @12892ms
[INFO] Started Jetty Server
[ForkJoinPool.commonPool-worker-3] INFO com.vaadin.flow.server.frontend.FrontendTools - using '/usr/bin/npx --yes --quiet --ignore-existing pnpm@5.18.10' for frontend package installation
npx: the --ignore-existing argument has been removed.
See `npm help exec` for more information
[ForkJoinPool.commonPool-worker-3] INFO com.vaadin.flow.server.frontend.TaskUpdatePackages - Frontend dependencies resolved successfully.
[ForkJoinPool.commonPool-worker-3] INFO com.vaadin.flow.server.frontend.TaskCopyFrontendFiles - Copying frontend resources from jar files ...
[ForkJoinPool.commonPool-worker-3] INFO com.vaadin.flow.server.frontend.TaskCopyFrontendFiles - Visited 18 resources. Took 28 ms.
[ForkJoinPool.commonPool-worker-3] INFO com.vaadin.base.devserver.AbstractDevServerRunner - Starting Webpack

------------------ Starting Frontend compilation. ------------------
[ForkJoinPool.commonPool-worker-3] INFO com.vaadin.base.devserver.AbstractDevServerRunner - Running Webpack to compile frontend resources. This may take a moment, please stand by...
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - [webpack-dev-server] Project is running at:
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - [webpack-dev-server] Loopback: http://localhost:43467/
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - [webpack-dev-server] On Your Network (IPv4): http://192.168.0.17:43467/
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - [webpack-dev-server] Content not from webpack is served from '/home/mavi/work/vaadin/starters/14/skeleton-starter-flow/target/classes/META-INF/VAADIN/webapp, /home/mavi/work/vaadin/starters/14/skeleton-starter-flow/src/main/webapp' directory
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - /home/mavi/work/vaadin/starters/14/skeleton-starter-flow/node_modules/.pnpm/esbuild@0.12.29/node_modules/esbuild/bin/esbuild:2
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - throw new Error(`esbuild: Failed to install correctly
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - ^
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - 
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - Error: esbuild: Failed to install correctly
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - 
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - Make sure you don't have "ignore-scripts" set to true. You can check this with
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - "npm config get ignore-scripts". If that returns true you can reset it back to
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - false using "npm config set ignore-scripts false" and then reinstall esbuild.
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - 
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - If you're using npm v7, make sure your package-lock.json file contains either
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - "lockfileVersion": 1 or the code "hasInstallScript": true. If it doesn't have
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - either of those, then it is likely the case that a known bug in npm v7 has
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - corrupted your package-lock.json file. Regenerating your package-lock.json file
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - should fix this issue.
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - 
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker -     at Object.<anonymous> (/home/mavi/work/vaadin/starters/14/skeleton-starter-flow/node_modules/.pnpm/esbuild@0.12.29/node_modules/esbuild/bin/esbuild:2:7)
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker -     at Module._compile (internal/modules/cjs/loader.js:999:30)
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker -     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker -     at Module.load (internal/modules/cjs/loader.js:863:32)
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker -     at Function.Module._load (internal/modules/cjs/loader.js:708:14)
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker -     at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:60:12)
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker -     at internal/main/run_main_module.js:17:47
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - 
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - ERROR in ./generated/vaadin.ts
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - Module build failed (from ../node_modules/.pnpm/esbuild-loader@2.15.1_webpack@4.46.0/node_modules/esbuild-loader/dist/index.js):
[dev-server-output] INFO com.vaadin.base.devserver.DevServerOutputTracker - Error: The service was stopped
...
```

**Comment by mvysny** [comment]
Workaround is to enable ignore-scripts as the error message suggests:

```
npm config set ignore-scripts false
mvn vaadin:dance
rm -rf node_modules
mvn -C clean package -Pproduction
```

However that will open a gaping security issue on your system since npm will basically download and run scripts from the internet.

**Comment by mvysny** [comment]
Hoorray, seems to be fixed in Vaadin 23.0.0.alpha1
